### PR TITLE
s3/test: remove the tempdir if test succeeds

### LINKED ISSUE #14690: tests: Jenkins: archiving artifacts takes >1 hour
Example from today:
```
15:18:26  [25878/25878] topology_raft_disabled  debug  [ PASS ] topology_raft_disabled.test_raft_upgrade_no_schema.3 42.25s
15:18:26  ------------------------------------------------------------------------------
15:18:26  CPU utilization: 42.8%
15:18:26  [Pipeline] echo
15:18:27  Uploading tests logs (x86_64)
15:18:27  [Pipeline] echo
15:18:27  Going to archive artifacts (as Jenkins artifacts): |testlog/**| from dir |/jenkins/workspace/scylla-master/next/scylla|, excluding |testlog/**/scylla-*/commitlog/**, testlog/**/scylla-*/data/**, testlog/**/scylla-*/hints/**, testlog/**/scylla-*/view_hints/**|, dryRun: false
15:18:27  [Pipeline] fileExists
15:18:27  [Pipeline] dir
15:18:27  Running in /jenkins/workspace/scylla-master/next/scylla
15:18:27  [Pipeline] {
15:18:27  [Pipeline] findFiles
16:26:39  [Pipeline] archiveArtifacts
16:26:39  Archiving artifacts
16:26:43  [Pipeline] }
16:26:43  [Pipeline] // dir
16:26:43  [Pipeline] echo
16:26:43  Going to archive artifacts (as Jenkins artifacts): |output_test_x86_64.txt| from dir |/jenkins/workspace/scylla-master/next/scylla|, excluding ||, dryRun: false
16:26:43  [Pipeline] fileExists
16:26:43  [Pipeline] dir
16:26:43  Running in /jenkins/workspace/scylla-master/next/scylla
```

have we started generating some large artifacts? Or is it a problem with the Jenkins pipeline?

cc @yaronkaikov @benipeled

### PR DESCRIPTION
**Pull request by tchaikov**
in 46616712, we tried to keep the tmpdir only if the test failed, and keep up to 1 of them using the recently introduced option of `tmp_path_retention_count`. but it turns out this option is not supported by the pytest used by our jenkins nodes, where we have pytest 6.2.5. this is the one shipped along with fedora 36.

so, in this change, the tempdir is removed if the test completes without failures. as the tempdir contains huge number of files, and jenkins is quite slow scanning them. after nuking the tempdir, jenkins will be much faster when scanning for the artifacts.

Fixes #14690
Signed-off-by: Kefu Chai <kefu.chai@scylladb.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
> CI state **`FAILURE`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/2607/

known issue, tracked by https://github.com/scylladb/scylladb/issues/14774

File: test/object_store/conftest.py [hunk 1]
```diff
@@ -44,12 +45,28 @@ def ssl(request):
     yield request.config.getoption('--ssl')
 
 
+@contextmanager
+def _maybe_remove_tempdir_on_complete(request, tempdir):
```
Comment (xemul) [inline test/object_store/conftest.py:49]: The size of hacks around tempdir fixture is becoming ... rather large. Maybe drop the fixture and maintain tempdir by hand, like the cql-pytest's run_with_temporary_dir prototype did?

Comment (tchaikov) [reply test/object_store/conftest.py:49]: you mean using global registry for tracking the cleanup? i think it is a stepback. i can, however drop the `_maybe_remove_tempdir_on_complete` and move it into `test_tempdir()`, despite that they serve the different purposes.

Comment (xemul) [reply test/object_store/conftest.py:49]: meh... global registry is, probably, not much better. Let's try with this one

**Pull request closed without merging**